### PR TITLE
C_C++ Templates

### LINKED ISSUE #2381: [Templates in C++]:
### Description

Documentation along with some code examples on Templates in C++.

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by anirudh09041**
<hr>

## Description 📜

I created a issue for "Templates in C++" which i have completed.
I have added Templates.md file which contain documentation along with example of Function template
and Class template 
Fixes #2381  

<hr>

## Type of change 📝

- [X] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [X] C/CPP

<hr>

## Checklist ✅

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [X] I have performed a self-review of my own code or work.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] My changes generates no new warnings.
- [X] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by anirudh09041** [comment]
@Suvraneel  please have a look at my PR

File: C_CPP/Templates & Libraries/templates.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Templates in C++
+
+---
+
+## Why do we need templates and What actually are templates ?
+
```
Comment (vasu-1) [inline C_CPP/Templates & Libraries/templates.md:6]:
```suggestion

---

## Why do we need templates and What are templates ?


```

File: C_CPP/Templates & Libraries/templates.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+100 is greater
+
+a is greater
+
+---
+
+## Points to Remember
+
+- A class created using class template is called a _template class_
+- A function created using function template is called a _template function_
```
Comment (vasu-1) [inline C_CPP/Templates & Libraries/templates.md:102]:
```suggestion
- A class created using a class template is called a _template class_
- A function created using a function template is called a _template function_

```

File: C_CPP/Templates & Libraries/readme.md [hunk 3]
```diff
@@ -1 +1 @@
-
+[Templates_in_C++.md]
```
Comment (vasu-1) [inline C_CPP/Templates & Libraries/readme.md:1]:
```suggestion
[Templates_in_C++](./templates.md)
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
kindly do this changes..!

**Comment by anirudh09041** [comment]
I have made the changes.

**Comment by vasu-1** [comment]
Great !!! Happy contributing !!

Good to go for merge !
@Suvraneel

**Comment by Suvraneel** [comment]
@anirudh09041 Has some stuff from the following link :
https://www.indcareer.com/paper/c-plus-plus-two-marks-part-i-0?amp=1

So, try to restructure/rephrase some sentences a little bit (eg, Templates are defined with a parameter that would be replaced by

the specified data type at the time of actual use of the class or function.)



Other than that we'd love it if it were a bit more detailed (if possible)

**Comment by anirudh09041** [comment]
Sure, I will make those changes as soon as possible

**Comment by Suvraneel** [comment]
@anirudh09041 Actually we reviewed and found that all content except the code are available elsewhere...
So no offence, looking at the size, quality & content, I need to close this PR... and transferring to #2584

Kindly take up a new issue & next time research 1st & write in your own words...
Thanks...

**Comment by Suvraneel** [comment]
Plagiarised from here mostly : https://informatica-interview.blogspot.com/2011/

**Pull request closed without merging**